### PR TITLE
Default group

### LINKED ISSUE #177: Feature: Default Group for self registered users
I couldn't find a way to set a default group for users who are self registering.

I'd like to invite a couple of people to my wishlist but don't want to ask everyone for their email addresses. When they register on their own without an invite link they get to a page which says "Can't find your group".

I could add them to the corresponding group afterwards but it would be great to set the default group in advance.

Thanks for your work and your precious time!

### PR DESCRIPTION
**Pull request by albinmedoc**
Closes #177

### CODE REVIEW

**Comment by albinmedoc** [comment]
When this is merged, I will start working on Proxy Header authentication.

**Comment by albinmedoc** [comment]
> When this is merged, I will start working on Proxy Header authentication.

I have to code ready for this as well. Will make a PR when this is merged, since it's based on this.

**Comment by cmintey** [comment]
I'm back and forth between this being a setting, or if there should be a `default` flag on the group model itself. If we do it the way you have it now, then there would need to be an additional check to make sure the group exists (in case the default group was deleted). 
If we put it on the group model, then we could query directly for the default group. There would need to be additional logic when setting the default group to make sure we don't have multiple, but I think that's fine. Then the flag would live on the group settings page, and it can be displayed in the group table as well.

**Comment by cmintey** [comment]
As for  your PR on proxy auth, can you open a separate feature request for that? I want to discuss it

**Comment by albinmedoc** [comment]
> As for your PR on proxy auth, can you open a separate feature request for that? I want to discuss it

Opened in #180

**Comment by albinmedoc** [comment]
I've committed a change that prevents the deletion of the default group. Since there can only be one default group, I believe the best approach is to store it globally and allow changes through the global settings. Additionally, I want to display the default group in the group table.

**Comment by cmintey** [comment]
Cool, I think that's appropriate. Functionality is good! I'm going to merge another PR which will cause some conflicts on this branch, just FYI

**Comment by albinmedoc** [comment]
I have resolved the conflicts

File: src/lib/components/admin/SettingsForm/DefaultGroup.svelte [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+<div class="flex flex-col space-y-2">
+    <h2 class="h2">Default group</h2>
+    <select id="defaultGroup" name="defaultGroup" class="select w-fit min-w-64" bind:value={groupId}>
+        <option value="">Select a group</option>
+        {#each groups as group}
+            <option value={group.id}>{group.name}</option>
+        {/each}
+    </select>
+</div>
```
Comment (cmintey) [inline src/lib/components/admin/SettingsForm/DefaultGroup.svelte:23]: In the PR I merged earlier, it added a new `BaseSetting` component you can use here

**Review by cmintey — CHANGES_REQUESTED** [review]
You should pass the groups to the `SettingsForm` in the `GlobalSettingsStep` component. You can pull the groups the same way as the config
```ts
let groups: Group[] = $state($page.data.groups);
```

**Comment by albinmedoc** [comment]
It should be fixed now

**Review by cmintey — APPROVED** [review]
Looks good. Lint check is failing, you can fix the formatting issues by running `pnpm format`

**Comment by albinmedoc** [comment]
I have now fixed the lint check

**Pull request merged**